### ISSUE TITLE
nightwatch example doesnt seem to work on my machine

### ISSUE THREAD

**Issue by posix4e** [meteor-velocity/velocity-examples#28]
I can't seem to get your nightwatch example work

Here's my app hanging
![screenshot from 2014-12-26 14 08 30](https://cloud.githubusercontent.com/assets/806363/5559604/b54673ec-8d08-11e4-9212-919b2ba1bbe1.png)

Here's how i ran the test
posix4e@posix4e-P27GV2:~/projects/velocity-examples/leaderboard-nightwatch$ meteor run --test --verbose
%% Parsing the --port option
[[[[[ ~/projects/velocity-examples/leaderboard-nightwatch ]]]]]

=> Started proxy.

=> Meteor 1.0.2.1: Performance and stability improvements to the command-line
   tool, and a new server console command: 'meteor shell'.

   This release is being downloaded in the background. Update your app to
   Meteor 1.0.2 by running 'meteor update'.

=> Started MongoDB.
I20141226-14:07:43.870(-8)? Lets register Nightwatch with Velocity...
=> Started your app.

=> App running at: http://localhost:3000/
I20141226-14:07:50.509(-8)? Server received request to run Nightwatch script...
I20141226-14:07:50.511(-8)? Installing Nightwatch bridge via Npm...
I20141226-14:07:53.172(-8)? nightwatch@0.5.3 node_modules/nightwatch
I20141226-14:07:53.172(-8)? ├── mkpath@0.1.0
I20141226-14:07:53.173(-8)? ├── ejs@1.0.0
I20141226-14:07:53.173(-8)? ├── minimatch@0.2.14 (sigmund@1.0.0, lru-cache@2.5.0)
I20141226-14:07:53.173(-8)? └── optimist@0.6.1 (wordwrap@0.0.2, minimist@0.0.10)
I20141226-14:07:53.173(-8)?
I20141226-14:07:53.173(-8)? Launching Nightwatch with JSON configuration file...
I20141226-14:07:53.183(-8)?
I20141226-14:07:53.183(-8)? Nightwatch exited.
I20141226-14:07:53.184(-8)? Parsing Nightwatch XML report files...
I20141226-14:07:53.185(-8)? process.env.PWD /home/posix4e/projects/velocity-examples/leaderboard-nightwatch
I20141226-14:07:53.185(-8)? testReportsPath /home/posix4e/projects/velocity-examples/leaderboard-nightwatch/tests/.reports/nightwatch-acceptance
I20141226-14:07:53.185(-8)? Parsing Nightwatch FIREFOX xml files...
^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A^[[A

**Comment by awatson1978** [comment]
Also, please make sure you are using [Firefox 31](https://ftp.mozilla.org/pub/mozilla.org/firefox/releases/31.0/).

**Comment by awatson1978** [comment]
This should be fixed, per pull-request #35.  If it continues to be a problem, please comment and we'll reopen.

**Comment by martinhbramwell** [comment]
Dear Abigail,

I am also having a miserable time getting this to work.  It simply hangs "Loading".

I wonder if you have "user tested" your documentation?

It took me a long time to realize Velocity was waiting for me to press the tiny gear icon, **then** press the blue "Run nightwatch tests" button.  Even so, then I had to wait for the browser to pop up,  **then** wait for the tests to complete,  **then** wait while it parses FIREFOX xml files, and wait.  I am still waiting. It is still flashing "Loading."  Does Nightwatch only report in the morning?

Is there some further action I have to take?

What other things do I have to install?  NPM?  Node.js? Java JRE? Jave JDK?  Oracle? OpenJDK?  Java for OSX exclusively?  Selenium Server?  Nightwatch?  Any particular versions?

Is installing enough?  Do I need to have them up and running first?  Wouldn't it be a lot faster if I did?

I see in the terminal, the line, "Starting selenium server... started - PID:  10080", but I then see it exit before I can type "ps aux | grep 10080" .  How can I tell if Selenium is dying for some reason?  Wrong Java version?


My log file before I click the blue "Run nightwatch tests" button : 

      yourself@BDD0:~/projects/velocity-examples/leaderboard-nightwatch$ meteor --verbose
      % Parsing the --port option
      [[[[[ ~/projects/velocity-examples/leaderboard-nightwatch ]]]]]

      => Started proxy.                             
      Local package version is up-to-date: blaze@2.0.4
      Local package version is up-to-date: random@1.0.2
      Local package version is up-to-date: mobile-status-bar@1.0.2
      Local package version is up-to-date: deps@1.0.6
      Local package version is up-to-date: velocity:nightwatch-framework@0.4.0
      Local package version is up-to-date: follower-livedata@1.0.3
      Local package version is up-to-date: spacebars@1.0.5
      Local package version is up-to-date: velocity:html-reporter@0.4.2
      Local package version is up-to-date: spacebars-compiler@1.0.4
      Local package version is up-to-date: launch-screen@1.0.1
      Local package version is up-to-date: http@1.0.10
      Local package version is up-to-date: json@1.0.2
      Local package version is up-to-date: check@1.0.4
      Local package version is up-to-date: retry@1.0.2
      Local package version is up-to-date: standard-app-packages@1.0.4
      Local package version is up-to-date: id-map@1.0.2
      Local package version is up-to-date: reactive-dict@1.0.5
      Local package version is up-to-date: mrt:moment@2.8.1
      Local package version is up-to-date: callback-hook@1.0.2
      Local package version is up-to-date: meteor@1.1.4
      Local package version is up-to-date: fastclick@1.0.2
      Local package version is up-to-date: minifiers@1.1.3
      Local package version is up-to-date: webapp@1.1.6
      Local package version is up-to-date: ejson@1.0.5
      Local package version is up-to-date: base64@1.0.2
      Local package version is up-to-date: url@1.0.3
      Local package version is up-to-date: blaze-tools@1.0.2
      Local package version is up-to-date: ddp@1.0.14
      Local package version is up-to-date: observe-sequence@1.0.4
      Local package version is up-to-date: tinytest@1.0.4
      Local package version is up-to-date: reactive-var@1.0.4
      Local package version is up-to-date: webapp-hashing@1.0.2
      Local package version is up-to-date: mongo@1.0.11
      Local package version is up-to-date: htmljs@1.0.3
      Local package version is up-to-date: ui@1.0.5 
      Local package version is up-to-date: amplify@1.0.0
      Local package version is up-to-date: meteor-platform@1.2.1
      Local package version is up-to-date: ordered-dict@1.0.2
      Local package version is up-to-date: session@1.0.5
      Local package version is up-to-date: livedata@1.0.12
      Local package version is up-to-date: templating@1.0.11
      Local package version is up-to-date: binary-heap@1.0.2
      Local package version is up-to-date: tracker@1.0.5
      Local package version is up-to-date: autoupdate@1.1.5
      Local package version is up-to-date: html-tools@1.0.3
      Local package version is up-to-date: reload@1.1.2
      Local package version is up-to-date: less@1.0.12
      Local package version is up-to-date: application-configuration@1.0.4
      Local package version is up-to-date: underscore@1.0.2
      Local package version is up-to-date: routepolicy@1.0.4
      Local package version is up-to-date: velocity:core@0.4.5
      Local package version is up-to-date: insecure@1.0.2
      Local package version is up-to-date: geojson-utils@1.0.2
      Local package version is up-to-date: minimongo@1.0.6
      Local package version is up-to-date: clinical:nightwatch@1.6.0
      Local package version is up-to-date: jquery@1.11.3
      Local package version is up-to-date: boilerplate-generator@1.0.2
      Local package version is up-to-date: logging@1.0.6
      => Started MongoDB.                           
      I20150315-11:01:44.238(-4)? Lets register Nightwatch with Velocity...
      => Started your app.

      => App running at: http://localhost:3000/

This is the log after I click th blue "Run nightwatch tests" button : 

      I20150315-11:05:31.326(-4)? Server received request to run Nightwatch script...
      I20150315-11:05:31.331(-4)? Installing Nightwatch bridge via Npm...
      I20150315-11:05:35.691(-4)? nightwatch@0.5.36 node_modules/nightwatch
      I20150315-11:05:35.692(-4)? ├── mkpath@0.1.0
      I20150315-11:05:35.693(-4)? ├── ejs@0.8.8
      I20150315-11:05:35.693(-4)? ├── minimatch@0.2.14 (sigmund@1.0.0, lru-cache@2.5.0)
      I20150315-11:05:35.693(-4)? ├── optimist@0.6.1 (wordwrap@0.0.2, minimist@0.0.10)
      I20150315-11:05:35.693(-4)? └── grunt@0.4.5 (which@1.0.9, dateformat@1.0.2-1.2.3, eventemitter2@0.4.14, getobject@0.1.0, rimraf@2.2.8, colors@0.6.2, async@0.1.22, grunt-legacy-util@0.2.0, hooker@0.2.3, exit@0.1.2, lodash@0.9.2, nopt@1.0.10, coffee-script@1.3.3, underscore.string@2.2.1, iconv-lite@0.2.11, glob@3.1.21, findup-sync@0.1.3, grunt-legacy-log@0.1.1, js-yaml@2.0.5)
      I20150315-11:05:35.693(-4)? 
      I20150315-11:05:35.694(-4)? Launching Nightwatch with JSON configuration file...
      [sudo] password for yourself: 

        [[ after a few seconds the browser pops up  ]]
        
Finally, this the log after the browser closes :    

      I20150315-11:06:34.031(-4)? 
      I20150315-11:06:34.031(-4)? [Walkthrough] Test Suite
      I20150315-11:06:34.031(-4)? ========================
      I20150315-11:06:34.031(-4)? 
      I20150315-11:06:34.032(-4)? Running:  Basic Nightwatch Walkthrough 
      I20150315-11:06:34.032(-4)? 
      I20150315-11:06:34.032(-4)? 
      I20150315-11:06:34.032(-4)? Nightwatch exited.
      I20150315-11:06:34.032(-4)? Running nightwatch/parse/xml...
      I20150315-11:06:34.032(-4)? process.env.PWD /home/yourself/projects/velocity-examples/leaderboard-nightwatch
      I20150315-11:06:34.032(-4)? testReportsPath /home/yourself/projects/velocity-examples/leaderboard-nightwatch/tests/.reports/nightwatch-acceptance
      I20150315-11:06:34.032(-4)? Parsing FIREFOX xml files...

But, in my web page it is still hanging at "Loading . . . "


Am I suppose to run as root?  When I click the blue "Run nightwatch tests" button, I see in the terminal a request for my *sudo* password.  That does not seem normal? It's especially weird to be required to do it every time I run the tests in a new terminal session.  Can't it install once and for all?

I sure hope some fellow Ubuntu user can look at this and give me some suggestions.

I am running a Xubuntu Desktop virtual machine guest under KVM.

      java version "1.7.0_75"
      OpenJDK Runtime Environment (IcedTea 2.5.4) (7u75-2.5.4-1~trusty1)
      OpenJDK 64-Bit Server VM (build 24.75-b04, mixed mode)

     npm --version
        1.3.10

     meteor --version
        1.0.3.2

**Comment by ih** [comment]
I'm also having it hang at

Parsing FIREFOX xml files...

**Comment by robwatkin** [comment]
Me too. (just so that I get any notifications) :-)

**Comment by awatson1978** [comment]
We're getting ready for a big update!  Stay tuned!  :)

**Comment by martinhbramwell** [comment]
How do we follow news on this?  Twitter?  forums.meteor.com?  FBI "10 most wanted"?

**Comment by awatson1978** [comment]
forums.meteor.com is a good place for it, I suppose.  

Long story short, we finally got Nightwatch integrated not just with Velocity, but with the ``meteor --test`` command.  Which means that the ``run_nightwatch.sh`` script is no longer needed (although it can still be used, if you like that workflow).  Along the way, we've also fixed a bunch of permission issues, gotten custom command and assertions working, and wrote a nightwatch-reporter package (which can bootstrap default test files for a project, getting people up to speed that much quicker).  

We got the core integration working last week, and I suspect it will be published tomorrow.  :fingerscrossed:
